Remove old signing key from securedrop-admin

eloquence opened the pull request:
## Status

Ready for review

## Description of Changes

- Ensures the previous signing key no longer validates tags verified via `securedrop-admin`. It's still valid for old  tags, but `securedrop-admin` should never have to verify those.
- Also removes the previously used UIDs.
- Updates GPG test output to current year for clarity (the fingerprint and year combinations otherwise don't make sense); it makes no difference in terms of test results. Also updates copyright statement for consistency.
- Sprinkles in some code comments :sparkles: 

Fixes #6511

## Testing

- On an Admin Workstation, apply the changes to `__init.py__` in this PR (I manually applied them to the 2.4.1 tag, but testing on this branch should work as well for this purpose) and bounce the network. Verify that you can update to the latest tag successfully.

## Deployment considerations

- This PR removes the UID variant without an email address as valid. I believe this should be safe, but would very much appreciate additional fact-checking on this, see my comment here: https://github.com/freedomofpress/securedrop/pull/6519#issuecomment-1220113645


## Checklist

- [x] Linting and tests (`make -C admin test`) pass in the admin development container
- [x] I stepped through test plan above.

Commit:
Remove old signing key from securedrop-admin

- Ensure the previous signing key no longer validates tags
  verified via `securedrop-admin`. It's still valid for old
  tags, but `securedrop-admin` should never have to verify
  those.
- Updates GPG test output to current year for clarity (the
  fingerprint and year otherwise don't make sense); it makes
  no difference in terms of test results.

legoktm commented on admin/securedrop_admin/__init__.py:
```diff
@@ -1021,7 +1019,7 @@ def update(args: argparse.Namespace) -> int:
         # match from good_sig_text[]
         if (
-            (RELEASE_KEYS[0] in gpg_lines[1] or RELEASE_KEYS[1] in gpg_lines[1])
+            RELEASE_KEYS[0] in gpg_lines[1]
```
Maybe `any(key in gpg_lines[1] for key in RELEASE_KEYS)`?

eloquence replied:
SGTM, done!

legoktm reviewed (commented):
https://github.com/freedomofpress/securedrop/blob/a17aeec4c9120cf6939b48145ebe4df1be9919a6/admin/securedrop_admin/__init__.py#L1007

Don't we also want to remove the securedrop-release-key@ from being valid here?

Commit:
Test against all supported release keys

Commit:
Remove old UIDs

eloquence commented:
> Don't we also want to remove the securedrop-release-key@ from being valid here?

Oh, great catch - I think both of those UIDs can indeed be safely removed as well. Pushed a commit to do so.

Commit:
Update test strings to include UID for current key

eloquence commented:
OK, I've removed those old UIDs in e8ae44aa8827854ef17ea36354149eedf3f08f06 and updated the associated tests in e7e589e923bcefe2da6d66dc126384aaca45b4aa.

My understanding is that the history of the UID is as follows:
- Up until we migrated to keys.openpgp.org in July 2019, there was no email address associated with the release key.
- keys.openpgp.org requires an email address, which is why we added securedrop-release-key@freedom.press.
- For the key rotation in 2021, we used securedrop-release-key-2021@freedom.press to avoid clashes.

All of which is to say that for pre-2021 keys, we can expect both versions with and without the email address to exist. For the 2021 key, only the version with an email address should exist.

For that reason, I think it's indeed safe to remove variants without an email address from the output checks. 

However, please fact-check me on this (especially @zenmonkeykstop). If there are circumstances where users might encounter output for the 2021+ key without the email address, the signature check would now fail.

zenmonkeykstop reviewed (approved):
Test plan checks out, agree with the reasoning around UIDs.
